### ISSUE TITLE
a11y audit: OverflowMenu a11y fixes

### ISSUE THREAD

**Issue by marijohannessen** [carbon-design-system/carbon-components-react#618]
### Detailed description

- [x] An element with WAI-ARIA role menuitem is not contained in or owned by an element with one of the following WAI-ARIA roles: group, menu, menubar. See rule help for remediation guidance.
![screen shot 2018-02-15 at 2 38 39 pm](https://user-images.githubusercontent.com/5447411/36279899-33423e88-125e-11e8-91d7-57e8a948762b.png)
![screen shot 2018-02-15 at 2 41 03 pm](https://user-images.githubusercontent.com/5447411/36279923-4ff4ba6a-125e-11e8-8fa2-b3e4cf0bcb91.png)

- [x] For the FloatingMenu variation: A WAI-ARIA widget must have an accessible name (see message help text for details)
![screen shot 2018-02-15 at 2 43 41 pm](https://user-images.githubusercontent.com/5447411/36280096-d71ead48-125e-11e8-9df0-161d1152dd12.png)

- [x] Cannot use keyboard interaction to navigate up and down the Overflow Menu using the up and down arrows
